From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Doc <nachito94@msn.com>
Date: Sun, 6 Oct 2024 11:04:15 -0300
Subject: [PATCH] Configurable bypass for anvil max level enchantment
 restriction

If in anvil you try enchant with a level over the max the anvil use the max level for the result item ignoring the level passed, this PR make this configurable

diff --git a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
index d685511104ac552dfc9ae2111e1bfb60fa812102..c4808bb2b6f0e12f607c089d6ce9e4b9990b6edc 100644
--- a/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
+++ b/src/main/java/net/minecraft/world/inventory/AnvilMenu.java
@@ -231,7 +231,7 @@ public class AnvilMenu extends ItemCombinerMenu {
                             flag2 = true;
                         } else {
                             flag1 = true;
-                            if (i2 > enchantment.getMaxLevel()) {
+                            if (i2 > enchantment.getMaxLevel() && !io.papermc.paper.configuration.GlobalConfiguration.get().misc.allowAnvilApplyEnchantmentOverMaxLevel) { // Paper - Allow by config ignore this override for max enchantment level
                                 i2 = enchantment.getMaxLevel();
                             }
 
